working/333 - shared memory multithreading/proposal.md

```diff
@@ -678,17 +685,28 @@ associated with that:
 
 In _shared **everything** multithreading_ world `callback` can be allowed to
 capture arbitrary state, however in _shared **native memory** multithreading_
-this state has to be restricted to trivially shareable types:
+this state has to be restricted to trivially shareable types. To make it
+completely unambigious we impose an additional requirement that all variables
+captured by a closure will need to be annotated with `@pragma('vm:shared')`:
+
 
 ```dart
-// This code is okay because `int` is trivially shareable.
+// This code is okay because the variable is annotated and `int` is
+// trivially shareable.
+@pragma('vm:shared') int counter = 0;
```

Comment:
nit: formatter will put a newline after the pragma. Maybe prefer Dart code formatting in the way the formatter would format it.

(ditto down below.)